src/Microsoft.Framework.Runtime.Hosting/Json/JsonDeserializer.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.Framework.Runtime.Json
+{
+    internal class JsonDeserializer
+    {
+        private JsonBuffer _buffer;
+
+        public JsonValue Deserialize(TextReader reader)
+        {
+            if (reader == null)
+            {
+                throw new ArgumentNullException(nameof(reader));
+            }
+
+            _buffer = new JsonBuffer(reader);
+            return Deserialize();
+        }
+
+        private JsonValue Deserialize()
+        {
+            var result = DeserializeInternal(_buffer.Read());
+
+            // There are still unprocessed char. The parsing is not finished. Error happened.
+            var nextToken = _buffer.Read();
+            if (nextToken.Type != JsonTokenType.EOL)
+            {
+                throw new JsonDeserializerException("Failed to continue deserializing. Additional token " + nextToken.Value, nextToken);
+            }
+
+            return result;
+        }
+
+        private JsonValue DeserializeInternal(JsonToken next)
+        {
+            if (next.Type == JsonTokenType.EOL)
+            {
+                return null;
+            }
+
+            if (next.Type == JsonTokenType.LeftSquareBracket)
+            {
+                return DeserializeArray(next);
+            }
+
+            if (next.Type == JsonTokenType.LeftCurlyBracket)
+            {
+                return DeserializeObject(next);
+            }
+
+            if (next.Type == JsonTokenType.String)
+            {
+                return new JsonString(next.Value, next.GetPosition());
+            }
+
+            if (next.Type == JsonTokenType.True || next.Type == JsonTokenType.False)
```

Comment:
I find this easier to consume and save allocating additional string to store the boolean value.